src/us/kbase/workspace/database/provenance/CreditMetadata.java

```diff
@@ -0,0 +1,345 @@
+	private final List<FundingReference> funding;
+	private final List<PermanentID> relatedIdentifiers;
+	private final List<Title> titles;
+
+	private static final ResourceType DEFAULT_RESOURCE_TYPE = ResourceType.DATASET;
+
+	private CreditMetadata(
+			final String identifier,
+			final String comments,
+			final String license,
+			final String version,
+			final ResourceType resourceType,
+			final List<Contributor> contributors,
+			final List<EventDate> dates,
+			final List<FundingReference> funding,
+			final List<PermanentID> relatedIdentifiers,
+			final List<Title> titles) {
+		this.comments = comments;
+		this.contributors = contributors;
+		this.dates = dates;
+		this.funding = funding;
+		this.identifier = identifier;
+		this.license = license;
+		this.relatedIdentifiers = relatedIdentifiers;
+		this.resourceType = resourceType;
+		this.titles = titles;
+		this.version = version;
+	}
+
+	public String getIdentifier() {
+		return identifier;
+	}
+
+	public Optional<String> getComments() {
+		return Optional.ofNullable(comments);
+	}
+
+	public Optional<String> getLicense() {
+		return Optional.ofNullable(license);
+	}
+
+	public Optional<String> getVersion() {
+		return Optional.ofNullable(version);
+	}
+
+	public ResourceType getResourceType() {
+		return resourceType;
+	}
+
+	public List<Contributor> getContributors() {
+		return Common.getList(contributors);
+	}
+
+	public List<EventDate> getDates() {
+		return Common.getList(dates);
+	}
+
+	public List<FundingReference> getFunding() {
+		return Common.getList(funding);
+	}
+
+	public List<PermanentID> getRelatedIdentifiers() {
+		return Common.getList(relatedIdentifiers);
+	}
+
+	public List<Title> getTitles() {
+		return Common.getList(titles);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(comments, contributors, dates, funding, identifier, license, relatedIdentifiers,
+				resourceType,
+				titles,
+				version);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		CreditMetadata other = (CreditMetadata) obj;
+		return Objects.equals(comments, other.comments)
+				&& Objects.equals(contributors, other.contributors)
+				&& Objects.equals(dates, other.dates)
+				&& Objects.equals(funding, other.funding)
+				&& Objects.equals(identifier, other.identifier)
+				&& Objects.equals(license, other.license)
+				&& Objects.equals(relatedIdentifiers, other.relatedIdentifiers)
+				&& Objects.equals(resourceType, other.resourceType)
+				&& Objects.equals(titles, other.titles)
+				&& Objects.equals(version, other.version);
+	}
+
+	/**
+	 * Gets a builder for the {@link CreditMetadata}.
+	 *
+	 * @param identifier unique persistent ID for the resource
+	 * @param resourceType type of the resource, as a {@link ResourceType}
+	 * @param contributors list of {@link Contributor} objects
+	 * @param titles list of {@link Title} objects
```

Comment:
Should mention that at least one title / contributor is required, here and below

Reply:
:+1: